Coding/fw/lib/WebServer/CaptivePortal.cpp

```diff
@@ -0,0 +1,172 @@
+/* INCLUDES ***************************************************************************************/
+#include <CaptivePortal.h>
+#include <Web_Config.h>
+#include <SPIFFS.h>
+
+/* C-Interface ************************************************************************************/
+extern "C"
+{
+}
+
+/* CONSTANTS **************************************************************************************/
+
+/* MACROS *****************************************************************************************/
+
+/* TYPES ******************************************************************************************/
+static void reqRestart();                                    /*< Changes the value of restartRequested to True*/
+static void credentialsProcessor(String name, String value); /*< Processor to save the required credentials */
+
+/**
+* Captive portal request handler.
+*/
+class CaptiveRequestHandler : public AsyncWebHandler
+{
+public:
+    /**
+    * Constructs the captive portal request handler.
+    */
+    CaptiveRequestHandler()
+    {
+    }
+
+    /**
+    * Destroys the captive portal request handler.
+    */
+    ~CaptiveRequestHandler()
+    {
+    }
+
+    /**
+    * Checks whether the request can be handled.
+    * @param[in] request   Web request
+    * @return If request can be handled, it will return true otherwise false.
+    */
+    bool canHandle(AsyncWebServerRequest *request) override
+    {
+        /* The captive portal handles every request. */
+        return true;
+    }
+
+    /**
+    * Handles the request.
+    * @param[in] request   Web request, which to handle.
+    */
+    void handleRequest(AsyncWebServerRequest *request) override
+    {
+        if (nullptr == request)
+        {
+            return;
+        }
+        /** Forces authentication */
+        if (!request->authenticate(WebConfig::getWEB_USER().c_str(), WebConfig::getWEB_PASS().c_str()))
+        {
+            return request->requestAuthentication();
+        }
+
+        if (HTTP_POST == request->method())
+        {
+            if (request->args() != 0)
+            {
+                int params = request->params();
+                for (int i = 0; i < params; i++)
+                {
+                    AsyncWebParameter *p = request->getParam(i);
+                    credentialsProcessor(p->name(), p->value());
+                }
+                request->send(SPIFFS, "/setCredentials.html", String(), false, captivePageProcessor);
+                reqRestart();
+            }
+        }
+        else if (HTTP_GET == request->method())
+        {
+            request->send(SPIFFS, "/STACredentials.html", String(), false, captivePageProcessor);
+        }
+        else
+        {
+            request->send(400, "text/html", "\n Error. Bad Request");
+        }
+    }
+
+    bool isRequestHandlerTrivial() override
+    {
+        return false;
+    }
+
+private:
+    static String captivePageProcessor(const String &var)
```

Comment:
Comment me please. Why does this return a default constructed string an ignores the input.

Reply:
It was there in case the captive portal contained Templates that needed to be processed. At the end, his was not the case, therefore I will delete it.